Features/hdf 8 view stats per pool

schystz opened the pull request:
## Description of Changes

Display the following stats per pool on Pools page:
- total pool value ($)
- BPT balance
- BPT staked value ($)
- HALO rewards earned

Other changes/fixes:
- fixed a bug in `develop` branch when approving the stake amount

Under the hood changes:
- updated halo hooks to consolidate all useful info in `poolsInfo` state (from `useBalancer` hook), removing all redundant logic from other hooks (e.g. `usePoolSummary`), making the app more efficient
- created a currency formatter helper to easily display currency amounts on the UI

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-8)


### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/SOOW-XXX-description`
* [x] All files have appropriate file headers and documentation
* ~~[ ] I have written new tests for your core changes, as applicable~~
* ~~[ ] I have successfully ran tests locally~~

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] Code style confirms to eslint/flow rules
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

tracyarciaga commented:
@schystz trying to test this feature branch, but the when I click the "Stake" on the main  nav bar, it just redirects to pool page.

schystz commented:
> @schystz trying to test this feature branch, but the when I click the "Stake" on the main nav bar, it just redirects to pool page.

@tracyarciaga it was behind develop thats why the stake tab isn't working (we merged Ernest's PR just yesterday). I have rebased from develop so both tabs should be working now.

ernestpascual reviewed (changes requested):
Hi @schystz sorry took time to verify some stuff:


Can we fix the HALO value to 2 decimal places? it seems to be breaking the UI. You can mint your own HALO from the scripts but i can send some as well.
<img width="978" alt="Screen Shot 2021-04-06 at 11 08 54 AM" src="https://user-images.githubusercontent.com/10893681/113658311-6105ca00-96d2-11eb-92ca-e164d1283453.png">

Im getting this errors, do you get this as well? this is when staking and unstaking BPT
<img width="1496" alt="Screen Shot 2021-04-06 at 2 09 03 PM" src="https://user-images.githubusercontent.com/10893681/113666327-b5647600-96e1-11eb-961d-a3b7b23a37e7.png">

schystz commented:
Thanks for the catch @ernestpascual ! Yeah I agree, HALO earnings need to be formatted to 2 decimal places, like what we are doing with BPT tokens.

I didn't notice the staking error before when I run it locally. I was able to stake and unstake even. But let me check again now to be sure.

schystz commented on src/halo-hooks/usePoolsSummary.ts:
```diff
@@ -1,36 +1,36 @@
 
   // Get totalSupply of each pool token
-  const totalSupplies = useTokenTotalSuppliesWithLoadingIndicator(poolTokens)[0]
+  const totalSupplies = useTokenTotalSuppliesWithLoadingIndicator(poolsAsTokens)[0]
```
Naming the variable `poolTokens` might be confused with `pool.tokens` (balancer pool tokens) so what I ended up doing instead is I created a new property `pool.asToken` then collectively called them `poolsAsTokens` (which represents an array of uniswap's `Token` object).

schystz replied:
<img width="197" alt="Screen Shot 2021-04-06 at 3 51 53 PM" src="https://user-images.githubusercontent.com/1032707/113677476-33c81480-96f0-11eb-9718-f2da0d0c178d.png">

Sharing this for further reference on the poolInfo object structure

ernestpascual replied:
thanks for the reference! agreeing with this as well.

schystz commented:
@ernestpascual updated with the following changes:

- refactored `usePoolsSummary` hook to move away from `useEffect` as it doesn't play nicely with `useBalances` and `useTokenTotalSuppliesWithLoadingIndicator` causing an endless loop. To achieve this, I used `useMemo` and created 3 new hooks to get: user total claimed HALO, user BPT staked per pool, user unclaimed HALO per pool (by using other uniswap hooks as pattern)

- added the 2 decimal place rounding when showing HALO balance

schystz commented on src/constants/index.ts:
```diff
@@ -83,6 +82,23 @@ export const BALANCER_SUBGRAPH_URL =
+    '0x': '0xccb0F4Cf5D3F97f4a55bb5f5cA321C3ED033f244'
+  }
+}
+
```
@tracyarciaga to give a better context - these are kovan addresses of popular tokens so we can map them out from coingecko prices (coingecko only keep tracks of mainnet addresses), which is a step required in able to display total pool value in the pools table. So yeah, we do need them here :) Feel free to suggest though if you think we should define this on another place.

ernestpascual reviewed (approved):
given the gas thing bug will be resolved on another issue, ill approve this pr @schystz  @tracyarciaga

schystz commented:
Thanks @tracyarciaga & @ernestpascual ! Merging this now to develop.

tracyarciaga reviewed (approved):
Tested, and the vesting modal flow is working as expected. 
Code changes looks good.
Just one minor thing before you @schystz merge:
1. Please rename DSRT to RNBW in the vesting modal.

ernestpascual reviewed (approved):
lgtm!
